rosco-manifests/src/main/java/com/netflix/spinnaker/rosco/manifests/kustomize/KustomizeTemplateUtils.java

```diff
@@ -0,0 +1,198 @@
+/*
+ * Copyright 2019 Armory, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.rosco.manifests.kustomize;
+
+import com.netflix.spinnaker.kork.artifacts.model.Artifact;
+import com.netflix.spinnaker.kork.web.exceptions.InvalidRequestException;
+import com.netflix.spinnaker.rosco.jobs.BakeRecipe;
+import com.netflix.spinnaker.rosco.manifests.TemplateUtils;
+import com.netflix.spinnaker.rosco.manifests.kustomize.mapping.Kustomization;
+import com.netflix.spinnaker.rosco.services.ClouddriverService;
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.io.FilenameUtils;
+import org.springframework.stereotype.Component;
+
+@Component
+@Slf4j
+public class KustomizeTemplateUtils extends TemplateUtils {
+  private final KustomizationFileReader kustomizationFileReader;
+
+  public KustomizeTemplateUtils(
+      KustomizationFileReader kustomizationFileReader, ClouddriverService clouddriverService) {
+    super(clouddriverService);
+    this.kustomizationFileReader = kustomizationFileReader;
+  }
+
+  public BakeRecipe buildBakeRecipe(
+      BakeManifestEnvironment env, KustomizeBakeManifestRequest request) {
+    BakeRecipe result = new BakeRecipe();
+    result.setName(request.getOutputName());
+    Artifact artifact = request.getInputArtifact();
+    if (artifact == null) {
+      throw new IllegalArgumentException("Exactly one input artifact must be provided to bake.");
+    }
+    String kustomizationfilename = FilenameUtils.getName(artifact.getReference());
+    if (kustomizationfilename != null
```

Comment:
Does it make sense to continue here if the file name is null, or should we throw in that case too?

Reply:
Yeah, it's probably safer to guard against a non well formed artifact than it is to assume we'll always be supplied a well formed artifact.